<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>880</width>
    <height>454</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Pattern Paint</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="patternToolbox" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">QWidget#patternToolbox {
  background:qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:0.9801 rgba(100,100,100, 255), stop:1 rgba(0, 0, 0, 255))
}</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>5</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="LetterboxScrollArea" name="scrollArea">
        <property name="palette">
         <palette>
          <active>
           <colorrole role="Base">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>255</red>
              <green>255</green>
              <blue>255</blue>
             </color>
            </brush>
           </colorrole>
           <colorrole role="Window">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>0</red>
              <green>0</green>
              <blue>0</blue>
             </color>
            </brush>
           </colorrole>
          </active>
          <inactive>
           <colorrole role="Base">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>255</red>
              <green>255</green>
              <blue>255</blue>
             </color>
            </brush>
           </colorrole>
           <colorrole role="Window">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>0</red>
              <green>0</green>
              <blue>0</blue>
             </color>
            </brush>
           </colorrole>
          </inactive>
          <disabled>
           <colorrole role="Base">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>0</red>
              <green>0</green>
              <blue>0</blue>
             </color>
            </brush>
           </colorrole>
           <colorrole role="Window">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>0</red>
              <green>0</green>
              <blue>0</blue>
             </color>
            </brush>
           </colorrole>
          </disabled>
         </palette>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="lineWidth">
         <number>0</number>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="PatternEditor" name="patternEditor">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>500</width>
           <height>387</height>
          </rect>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>500</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>500</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="baseSize">
          <size>
           <width>10</width>
           <height>10</height>
          </size>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>880</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_File"/>
    <addaction name="actionSave_File"/>
    <addaction name="separator"/>
    <addaction name="actionExport_pattern_for_Arduino"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionSystem_Information"/>
    <addaction name="separator"/>
    <addaction name="actionVisit_the_BlinkyTape_forum"/>
    <addaction name="actionTroubleshooting_tips"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionResize_Pattern"/>
    <addaction name="separator"/>
    <addaction name="actionFlip_Horizontal"/>
    <addaction name="actionFlip_Vertical"/>
    <addaction name="separator"/>
    <addaction name="actionClear_Pattern"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionSave_to_Tape"/>
    <addaction name="separator"/>
    <addaction name="actionAutomatically_connect"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_rainbow_sketch"/>
    <addaction name="actionAddress_programmer"/>
   </widget>
   <widget class="QMenu" name="menuInstruments">
    <property name="title">
     <string>Instruments</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuInstruments"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="tools">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>true</bool>
   </attribute>
   <addaction name="actionLoad_File"/>
   <addaction name="actionSave_File"/>
   <addaction name="actionSave_to_Tape"/>
   <addaction name="actionConnect"/>
   <addaction name="separator"/>
   <addaction name="actionPlay"/>
  </widget>
  <widget class="QToolBar" name="instruments">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>true</bool>
   </attribute>
  </widget>
  <action name="actionLoad_File">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/resources/images/open.png</normaloff>:/resources/images/open.png</iconset>
   </property>
   <property name="text">
    <string>Load File</string>
   </property>
   <property name="iconText">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Load Pattern</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_File">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/resources/images/save.png</normaloff>:/resources/images/save.png</iconset>
   </property>
   <property name="text">
    <string>Save File</string>
   </property>
   <property name="iconText">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionSystem_Information">
   <property name="text">
    <string>System Information</string>
   </property>
  </action>
  <action name="actionExport_pattern_for_Arduino">
   <property name="text">
    <string>Export pattern for Arduino</string>
   </property>
  </action>
  <action name="actionVisit_the_BlinkyTape_forum">
   <property name="text">
    <string>Visit the BlinkyTape forum</string>
   </property>
  </action>
  <action name="actionTroubleshooting_tips">
   <property name="text">
    <string>Troubleshooting tips</string>
   </property>
  </action>
  <action name="actionFlip_Horizontal">
   <property name="text">
    <string>Flip Horizontal</string>
   </property>
   <property name="toolTip">
    <string>Flip the Pattern in the horizontal direction</string>
   </property>
  </action>
  <action name="actionFlip_Vertical">
   <property name="text">
    <string>Flip Vertical</string>
   </property>
   <property name="toolTip">
    <string>Flip the Pattern in the vertical direction</string>
   </property>
  </action>
  <action name="actionLoad_rainbow_sketch">
   <property name="text">
    <string>Restore Rainbow Sketch</string>
   </property>
  </action>
  <action name="actionSave_to_Tape">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/resources/images/save_to_tape.png</normaloff>:/resources/images/save_to_tape.png</iconset>
   </property>
   <property name="text">
    <string>Save to Tape</string>
   </property>
  </action>
  <action name="actionClear_Pattern">
   <property name="text">
    <string>Clear Pattern</string>
   </property>
  </action>
  <action name="actionResize_Pattern">
   <property name="text">
    <string>Resize Pattern</string>
   </property>
  </action>
  <action name="actionAddress_programmer">
   <property name="text">
    <string>Address programmer</string>
   </property>
  </action>
  <action name="actionAutomatically_connect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Automatically connect</string>
   </property>
  </action>
  <action name="actionPlay">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/resources/images/play.png</normaloff>:/resources/images/play.png</iconset>
   </property>
   <property name="text">
    <string>Play</string>
   </property>
  </action>
  <action name="actionConnect">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/resources/images/connect.png</normaloff>:/resources/images/connect.png</iconset>
   </property>
   <property name="text">
    <string>Connect</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PatternEditor</class>
   <extends>QWidget</extends>
   <header>patterneditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LetterboxScrollArea</class>
   <extends>QScrollArea</extends>
   <header>letterboxscrollarea.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="images.qrc"/>
 </resources>
 <connections/>
</ui>
